pydm/widgets/archiver_time_plot.py

```diff
@@ -57,14 +60,27 @@ def __init__(self, channel_address: Optional[str] = None, use_archive_data: bool
 
     def to_dict(self) -> OrderedDict:
         """Returns an OrderedDict representation with values for all properties needed to recreate this curve."""
-        dic_ = OrderedDict(
-            [
-                ("useArchiveData", self.use_archive_data),
-            ]
-        )
+        dic_ = OrderedDict([("useArchiveData", self.use_archive_data), ("liveData", self.liveData)])
         dic_.update(super(ArchivePlotCurveItem, self).to_dict())
         return dic_
 
+    @property
+    def liveData(self):
+        return self._liveData
+
+    @liveData.setter
+    def liveData(self, get_live: bool):
```

Comment:
If we set `self._liveData = True` before the missing data can be inserted, then the live data buffer can get messed up.
This can happen when live data is fetched before the empty section in the buffer is filled. `np.roll` can be called on the live buffer at the wrong time and the trace reflects the wrong information.

We might want to set `_liveData = True` in an else block at the bottom of the setter to capture cases where the signal is not emitted.

Reply:
At least that should happen, otherwise as an example the checkbox PR won't work properly since it depends on the setter actually flipping `_liveData` to True when clicked

Reply:
Ok, I just tested out setting _liveData to True in the setter instead of the archiver slot. It worked perfectly. My other argument was outdated information from a few months ago when we wrote this :(

Reply:
We moved live data setting to true inside the setter and removed it from the buffer. Testing showed that this was safe.